src/piratechain/PiratechainEngine.ts

```diff
@@ -295,12 +300,14 @@ export class PiratechainEngine extends CurrencyEngine<
     this.birthdayHeight = piratechainPrivateKeys.birthdayHeight
 
     try {
-      this.synchronizer = await this.makeSynchronizer({
+      this.synchronizerPromise = this.makeSynchronizer({
         mnemonicSeed: piratechainPrivateKeys.mnemonic,
         birthdayHeight: piratechainPrivateKeys.birthdayHeight,
         alias: base16.stringify(base64.parse(this.walletId)),
         ...rpcNode
       })
+      this.synchronizer = await this.synchronizerPromise
+      this.synchronizerResolver(this.synchronizer)
```

Comment:
This logic was hard to understand, but I finally figured it out. Maybe it needs a comment:

1. Replace `this.synchronizerPromise` with a fresh promise. The old promise might have already been resolved.
2. People might be waiting on the old promise, so resolve that.